sdks/python/apache_beam/runners/portability/fn_api_runner.py

```diff
@@ -139,6 +139,25 @@ def done(self):
       self._state = self.DONE_STATE
 
 
+class _PartitionBuffer(object):
+  """This class is created to support partition(n) function."""
+  def __init__(self, inputs):
+    self._inputs = inputs
+
+  def partition(self, n):
+    v = list(self._inputs.values())[0]
+    if isinstance(v, list):
```

Comment:
I still think it is more natural to use polymorphism rather than if-elif chains for known types. Is there a reason to not do this?

Reply:
I added `_ListBuffer` class to support partitioning for list.